slack-lint/src/main/java/slack/lint/resources/MissingResourceImportAliasDetector.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2022 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package slack.lint.resources
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Context
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.LintFix
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import org.jetbrains.kotlin.psi.KtImportDirective
+import org.jetbrains.uast.UElement
+import org.jetbrains.uast.UImportStatement
+import org.jetbrains.uast.USimpleNameReferenceExpression
+import org.jetbrains.uast.getContainingUFile
+import org.jetbrains.uast.getQualifiedParentOrThis
+import slack.lint.resources.model.IMPORT_ALIASES
+import slack.lint.resources.model.RootIssueData
+import slack.lint.util.sourceImplementation
+
+/** Reports an error when an R class, other than the local module's, is imported without an import alias. */
+class MissingResourceImportAliasDetector : Detector(), SourceCodeScanner {
+
+  private val fixes = mutableListOf<LintFix>()
+  private var rootIssueData: RootIssueData? = null
+
+  override fun afterCheckFile(context: Context) {
+    rootIssueData?.let {
+      context.report(
+        ISSUE,
+        it.nameLocation,
+        "Please use an import alias here",
+        quickfixData =
+        fix()
+          .name("Add import alias")
+          // Apply the fixes in reverse so that the ranges/locations don't change.
+          .composite(*fixes.reversed().toTypedArray())
+          .autoFix()
+      )
+      rootIssueData = null
+      fixes.clear()
+    }
+  }
```

Comment:
Can you document this to explain what's happening here more? It's not totally clear at a glance why we're doing one big composite fix here vs attaching the fix to each report (which is the convention in lint). I suspect we don't even want to offer a fix in this case and instead want to suggest using the IDE inspection for this, which will do all this for them better than we can in a lint.

Reply:
We're doing one composite fix here because all the issues are related and we want them to be fixed in one go as seen [here](https://user-images.githubusercontent.com/3355288/192290006-883e21c3-5440-4b85-a2fe-b751c233c03c.mp4). If we attached each fix to its report, we'd end up with references to an alias that's not imported, or an import with an import alias while the simple name references still reference `R` and not the import alias.
It's also useful for automatically fixing the current issues via `lintFix` instead of baselining them.

Reply:
Documented this in 08691ee6e622c11376e8d34eac40cf8135251f5a.